src/types/qwbuffer.h

```diff
@@ -19,6 +19,26 @@ class QW_CLASS_OBJECT(buffer)
     QW_SIGNAL(release)
 
 public:
+    struct droper {
+        static inline void cleanup(qw_buffer *pointer) { if (pointer) pointer->drop(); }
+        void operator()(qw_buffer *pointer) const { cleanup(pointer); }
```

Comment:
why not just:

```
        void operator()(qw_buffer *pointer) const { if (pointer) pointer->drop(); }
```

Reply:
Both `cleanup` and `operator()` ensure it can also use with Qt's smart pointer.